common/src/main/java/com/wynntils/models/quests/QuestModel.java

```diff
@@ -143,26 +148,56 @@ private String stripPrefix(String name) {
         return StringUtils.replaceOnce(name, MINI_QUEST_PREFIX, "");
     }
 
-    void updateQuestsFromQuery(List<QuestInfo> newQuests) {
+    private void updateQuestsFromQuery(List<ContentInfo> newContent, List<StyledText> progress) {
+        List<QuestInfo> newQuests = new ArrayList<>();
+
+        for (ContentInfo content : newContent) {
+            if (content.type() != ContentType.QUEST && content.type() != ContentType.STORYLINE_QUEST) {
+                WynntilsMod.warn("Incorrect quest content type recieved: " + content);
+                continue;
+            }
+            QuestInfo questInfo = getQuestInfoFromContent(content);
+            newQuests.add(questInfo);
+        }
         quests = newQuests;
         WynntilsMod.postEvent(new QuestBookReloadedEvent.QuestsReloaded());
     }
 
-    void updateMiniQuestsFromQuery(List<QuestInfo> newMiniQuests) {
+    private void updateMiniQuestsFromQuery(List<ContentInfo> newContent, List<StyledText> progress) {
+        List<QuestInfo> newMiniQuests = new ArrayList<>();
+
+        for (ContentInfo content : newContent) {
+            if (content.type() != ContentType.MINI_QUEST) {
+                WynntilsMod.warn("Incorrect mini-quest content type recieved: " + content);
+                continue;
+            }
+            QuestInfo questInfo = getQuestInfoFromContent(content);
+            newMiniQuests.add(questInfo);
+        }
+
         miniQuests = newMiniQuests;
         WynntilsMod.postEvent(new QuestBookReloadedEvent.MiniQuestsReloaded());
     }
 
+    private static QuestInfo getQuestInfoFromContent(ContentInfo content) {
+        // We should always have a length, but if not, better fake one than crashing
+
+        return new QuestInfo(
+                content.name(),
+                QuestStatus.fromContentStatus(content.status()),
+                QuestLength.fromContentLength(content.length()),
+                content.requirements().level().key(),
+                content.description().orElse(StyledText.EMPTY),
+                // FIXME! Additional requirements missing
```

Comment:
Speaks for itself

Reply:
Yeah, I forgot about that. But even if I hadn't, I wouldn't have had the time to fix it. Tracked in https://github.com/Wynntils/Artemis/issues/1730.